android/build.gradle

```diff
@@ -9,13 +9,17 @@ buildscript {
 
 apply plugin: 'com.android.library'
 
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
```

Comment:
I don't get it why do you need it to be configurable through extensions.
The way gradle works is that you can define the "minimum" version of everything, if the root project contains a newer version, it will override the old one. That happens both to the SDK version and to dependencies

Reply:
first: Google will remove the `compile` option at the end of 2018, However, lots of 3rd do not upgrade them to `implementation` or `provide` etc. on time.
Second: lots of library module was used, we'd better to confirm to use the same configuration, such as miniSDK, targetSdk, and supportLibrary.

Reply:
I'm not talking about replacing `compile` to `implementation`, I agree with that.
I'm talking about the necessity of using an extension, where you can already override a version number through Gradle's hierarchy.

Reply:
add below code snippet to your build.gradle(the top build.gradle file)
```
ext {
    compileSdkVersion = 28
    buildToolsVersion = "27.0.3"
    supportVersion = "28.0.0"
    minSdkVersion = 19
}
```